packages/aws-cdk/test/cli/telemetry/collect-telemetry.test.ts

```diff
@@ -0,0 +1,35 @@
+import { Context } from '../../../lib/api/context';
+import { canCollectTelemetry } from '../../../lib/cli/telemetry/collect-telemetry';
+
+describe(canCollectTelemetry, () => {
+  let context: Context;
+
+  beforeEach(() => {
+    context = new Context();
+  });
+
+  test('returns true by default', async () => {
+    expect(canCollectTelemetry(context)).toBeTruthy();
+  });
+
+  test('returns false if env variable is set', async () => {
+    process.env.DISABLE_CLI_TELEMETRY = 'true';
```

Comment:
Should be in a finally block that reverts back to the original value (see if we already have a `withEnv` or something).